File: src/users-node/application/mapper/UserMapper.ts
import { UserResponseDto } from "../dto/response/UserResponseDto";

export class UserMapper {
  static toResponse(user: User): UserResponseDto {
    return { id: user.id, name: user.name, email: user.email };
  }

  static toDomain(dto: { name: string; email: string }): User {
    return { id: 0, name: dto.name, email: dto.email };
  }
}

File: src/users-java/src/main/java/com/users/application/mapper/UserMapper.java
package com.users.application.mapper;

import com.users.application.dto.request.UserRequestDto;
import com.users.application.dto.response.UserResponseDto;
import com.users.domain.model.UserModel;

public class UserMapper {

    public UserResponseDto toResponseDto(UserModel userModel) {
        return new UserResponseDto(userModel.getId(), userModel.getName(),  userModel.getEmail());
    }

    public UserModel toModel(UserRequestDto userRequestDto) {
        return new UserModel(userRequestDto.name(), userRequestDto.email());
    }

}
